huggingface/azureml/hf-ort.py

```diff
@@ -0,0 +1,164 @@
+    "bert-large" : 'language-modeling',
+    "distilbert-base" : 'language-modeling',
+    "gpt2" : 'language-modeling',
+    "bart-large" : 'translation',
+    "t5-large" : 'translation',
+    "deberta-v2-xxlarge" : 'text-classification',
+    "roberta-large" : 'question-answering'
+}
+
+CONFIG_ARGS_DICT = {
+    "pt-fp16" : [],
+    "ort" : ['--ort'],
+    "ds_s0" : ['--deepspeed', 'ds_config_zero_0.json'],
+    "ds_s0_ort" : ['--ort', '--deepspeed', 'ds_config_zero_0.json'],
+    "ds_s1" : ['--deepspeed', 'ds_config_zero_1.json'],
+    "ds_s1_ort" : ['--ort', '--deepspeed', 'ds_config_zero_1.json']
+}
+
+TRAINER_DIR = '../../huggingface-transformers/examples/pytorch'
+
+import os
+import shutil
+import sys
+import shlex
+from datetime import datetime
+import argparse
+
+# AzureML libraries
+import azureml.core
+from azureml.core import Experiment, Workspace, Datastore, Run, Environment
+from azureml.core.compute import ComputeTarget, AmlCompute, AksCompute
+from azureml.core.compute_target import ComputeTargetException
+from azureml.core import ScriptRunConfig
+from azureml.core.runconfig import PyTorchConfiguration
+
+# Check core SDK version number
+print("SDK version:", azureml.core.VERSION)
+
+print("The arguments are: " + str(sys.argv))
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("--gpu_cluster_name",
+                        help="GPU cluster for the script to run on", type=str, required=True)
+
+parser.add_argument("--hf_model",
+                        help="Huggingface models to run", type=str, required=True,
+                        choices=['bert-large', 'distilbert-base', 'gpt2', 'bart-large', 't5-large', 'deberta-v2-xxlarge', 'roberta-large'])
+
+parser.add_argument("--run_config",
+                        help="Run configuration indicating pytorch or ort, deepspeed stage", type=str, required=True,
+                        choices=['pt-fp16', 'ort', 'ds_s0', 'ds_s0_ort', 'ds_s1', 'ds_s1_ort'])
+
+parser.add_argument("--workspace_name",
+                        help="Name of the AzureML workspace", type=str, required=False)
+
+parser.add_argument("--resource_group",
+                        help="Resource group that AzureML workspace belongs to", type=str, required=False)
+
+parser.add_argument("--subscription_id",
+                        help="Subscription of that AzureML workspace belongs to", type=str, required=False)
+# model params 
+parser.add_argument("--model_batchsize",
+                        help="Model batchsize", type=int, required=False)
+
+parser.add_argument("--max_steps",
+                        help="Max step that a model will run", type=int, default=200, required=False)
+
+parser.add_argument("--process_count",
+                        help="Total number of GPUs (not GPUs per node)", type=int, required=False, default=8)
+
+parser.add_argument("--node_count",
+                        help="node count", type=int, required=False, default=1)
+
+args = parser.parse_args()                  
```

Comment:
Can we allow users to select datasets for the finetuning tasks

Reply:
We will add this support in future